src/components/source-switcher-button/source-switcher-button.js

```diff
@@ -0,0 +1,135 @@
+import videojs from 'video.js';
+import './source-switcher-button.scss';
+
+const MenuButton = videojs.getComponent('MenuButton');
+const MenuItem = videojs.getComponent('MenuItem');
+
+class SourceSwitcherButton extends MenuButton {
+  constructor(player, options = {}) {
+    super(player, options);
+
+    this.controlText(options.tooltip || 'Sources');
+    this._emptyLabel = options.emptyLabel || 'No sources';
+
+    this._items = options.items || [];
+    this._selectedIndex = Number.isInteger(options.defaultIndex) ? options.defaultIndex : undefined;
+
+    const onSelected = typeof options.onSelected === 'function' ? options.onSelected : null;
+
+    this._onSelected = onSelected || null;
+    this._setEnabled(Array.isArray(this._items) && this._items.length > 0);
+  }
+
+  buildCSSClass() {
+    const empty = !Array.isArray(this._items) || this._items.length === 0;
+    return `vjs-source-switcher-button${empty ? ' vjs-source-switcher-disabled' : ''} ${super.buildCSSClass()}`;
+  }
+
+  createItems() {
+    if (!Array.isArray(this._items) || this._items.length === 0) {
+      const empty = new MenuItem(this.player_, {
+        label: this._emptyLabel || 'No sources',
+        selectable: false
+      });
+      empty.addClass('vjs-source-switcher-empty');
+      empty.disable();
+      return [empty];
+    }
+
+    return this._items.map(({ label, value }, index) => {
+      const item = new MenuItem(this.player_, {
+        label,
+        selectable: true,
+        selected: index === this._selectedIndex
+      });
+      item.value = value;
+      item._ssIndex = index;
+
+      item.on('click', () => {
+        this.menu.children().forEach((child) => {
+          if (child instanceof MenuItem) {
+            child.selected(child._ssIndex === index);
+          }
+        });
+
+        this._selectedIndex = index;
+        const payload = { index, value, label };
+        if (this._onSelected) this._onSelected(payload, this.player_);
+        this.player_.trigger('sourceswitcher:change', payload);
+      });
+
+      return item;
+    });
+  }
+
+  setItems(items, keepSelection = true) {
```

Comment:
Why `keepSelection` defaults to false if its only usage needs `false`? looks like this can be simplified